enhancements/update/clusterversion-osus-signatures.md

```diff
@@ -0,0 +1,148 @@
+
+CVO currently fetches the upstream urls for signature verification through [cluster-update-keys]. 
+To improve disconnected cluster user experience we need to provide a way for the CVO  to get the required signatures 
+from the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images to update the clusters.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add a new signature store URI.
+* As a cluster administrator, I want the cluster's CVO to replace the existing signature store URIs with a user provided URI
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated in the process
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* The api extension will replace the existing stores. 
+* To use existing stores, admin will have to add them manually with locally hosted stores, although locally hosted stores 
+are not required for this functionality to work.
+* This will be an array as it'll allow the admin to use locally configured stores with the existing stores
+* Admins will still be able to use the existing config maps for signature verification. 
+
+### Risks and Mitigations
+
+* The responsibility of mirroring the right signatures is going to be same as mirroring the right payload as we expect 
+customers will use oc-mirror to mirror the release payload as well building the graph-data container image with the signatures.
+* Currently, customers use [oc-mirror] to mirror release payload and Cincinnati graph data information for OSUS running in 
+the disconnected environment.
+* [oc-mirror] already has the signature information when it mirrors the release payload. We will be reusing the same 
+image signatures to serve through the OSUS URI.
+
+### Drawbacks
+
+* The cluster admin will have to have their own due diligence to make sure that the release images and verification signatures are
+  authentic.
+
+## Design Details
+
+### Test Plan
+
+* CVO unit tests will be expanded and/or created to test the new logic.
```

Comment:
did we want to expand on what these tests will look like? I don't expect us to stand up a new signature service, but I guess that's possible.  This is complicated by the integration plugging in at a high-enough level where unit tests are hard.  And it's hard to write CI that depends on signature availability unless we restrict testing to post-candidate-channel updates, because before that, there aren't trusted ART signatures.  And while there is theoretically a CI signing key, it's not actually wired up to anything signing CI releases: openshift/origin#26073

Reply:
changed the block to say, existing tests will cover this.
we can add unit tests to check if CVO's default stores are replaced with custom ones.